src/kili/use_cases/utils.py

```diff
@@ -0,0 +1,91 @@
+"""Utils for use cases."""
+from itertools import chain
+from typing import Dict, List, Optional, cast
+
+from kili.adapters.kili_api_gateway import KiliAPIGateway
+from kili.adapters.kili_api_gateway.helpers.queries import QueryOptions
+from kili.core.utils import pagination
+from kili.domain.asset import AssetExternalId, AssetFilters, AssetId
+from kili.domain.asset.helpers import check_asset_identifier_arguments
+from kili.domain.project import ProjectId
+from kili.domain.types import ListOrTuple
+from kili.exceptions import NotFound
+
+
+class UseCasesUtils:
```

Comment:
We plan to do neither a use-case nor a gateway fonction, but a utility function. Here is a Use Case and it is too generic, I am afraid that other dev might add too many use case here as a "trash use case".
I would rather transform it into a `get_asset_ids_or_throw_error` and `infer_ids_from_external_ids` functions that takes `kili_api_gateway` as argument to truely have a utility function and not a utility use case

Reply:
and this utility could be in a module in the asset use case, not to have a module with every utility functions for all use cases

Reply:
no, this is not a use case, but the utils that can be used in the use cases. Can be functions as well, but it feels natural to declare the dependency in the class argument. Can be moved to an `AssetUseCasesUtils` eventually.

Reply:
I think that it will be cleaner in AssetUseCasesUtils

Reply:
done